scripts/helper_numpy.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# coding: utf8
+'''Compute a Numpy friendly data-structures from VGGish files
+Example
+-------
+$ cd {repo_root}
+$ python ./scripts/helper_numpy.py --csv_file /path/to/sparse-labels.csv \
+--vggish_path /path/to/vggish/ --output_file /path/to/output.npz
+
+Any valid output file '*.npz' has to be speficied by the user
+'''
+
+from __future__ import print_function
+import argparse
+import numpy as np
+import os
+import pandas as pd
+import sys
+import json
+
+
+def main(csvfile, vggishpath, outfile):
+
+    success = []
+    df = pd.read_csv(csvfile)
+    instruments = np.unique(df['instrument'])
+    sample_key = np.unique(df['sample_key'])
+    songs_num = len(sample_key)
+    inst_num = len(instruments)
+
+    print('Extracting the vggish features...')
+    X = np.empty([songs_num, 10, 128], dtype=int)
+    count = 0
+    for sk in sample_key:
+        sk_prefix = sk[:3]
+        full_name = os.path.join(vggishpath, sk_prefix, sk + '.json')
+        with open(full_name, 'r') as f:
+            X_tmp = json.load(f)
+        X[count] = X_tmp['features']
+        count += 1
+
+    print('Extracting the labels information...')
+    Y_true = 0.5 * np.ones([songs_num, inst_num], dtype=float)
+    Y_mask = np.zeros([songs_num, inst_num], dtype=bool)
+
+    inst_lab = np.copy(df)
+    for inst in inst_lab:
+        x_pos = int(np.arange(songs_num)[sample_key == inst[0]])
```

Comment:
I guess the integer-based indexing is fragile (there are other ways to iterate over dataframes), but this is probably fine

Reply:
This block seems pretty mysterious to me.  Why copy the dataframe?  Would it be possible to have a tuple with variable names rather than opaque `inst`?  e.g. `for (key, instrument, value) in inst_lab: ...`?

Reply:
what would you suggest?

Reply:
the last message was for @ejhumphrey 

@bmcfee re: copying the dataframe. It seemed a bit more efficient to me but I may not have used the most efficient implementation. The for loops takes about 15 seconds to complete with the numpy copy and about 18 seconds without it.
The implementation without it is:
```
    for _, (key, instrument, value) in df.iterrows():
        x_pos = int(np.arange(songs_num)[sample_key == key])
        y_pos = int(np.arange(inst_num)[instruments == instrument])
        Y_true[x_pos, y_pos] = value
        Y_mask[x_pos, y_pos] = 1
```
Would you suggest something different?
re: `(key, instrument, value)`, sure thing

Reply:
For this, I'd definitely favor readability over efficiency.

I think the slowness is coming from doing a bunch of arange constructions followed by mask indexing.  I'm not sure it's worth accelerating, but you might be able to do this more efficiently by pre-computing the sample-to-index mapping.

Reply:
The only issue a tuple with variable names is that if we add rows to the sparse label CSV file, as Eric is doing and as is discussed in #21, the script won't work. Could that be a problem?

Reply:
it's not efficient, but if you do something like this to preserve each row as a Series with attrs, e.g. 
```
for _, row in df.iterrows():
    x_pos = int(np.arange(songs_num)[sample_key == row.sample_key])
    ...
```

then column ordering doesn't matter, and you can add / shuffle columns to your heart's content

Reply:
👍